blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/HttpRequest.java

```diff
@@ -0,0 +1,135 @@
+package io.jenkins.blueocean.blueocean_github_pipeline;
+
+import com.fasterxml.jackson.databind.JsonMappingException;
+import io.jenkins.blueocean.commons.ServiceException;
+import org.apache.commons.io.IOUtils;
+import org.kohsuke.github.HttpConnector;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Array;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Collections;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * @author Vivek Pandey
+ */
+class HttpRequest {
```

Comment:
There isn't really anything specific to GitHub in it though:) Anyways its package private, so is fine IMO.